docs/lib/geo/fragments/js/search.md

```diff
@@ -0,0 +1,209 @@
+
+```bash
+npm install -S @maplibre/maplibre-gl-geocoder
+```
+
+If you haven't already add maplibre-gl-js and maplibre-gl-js-amplify to your app with `yarn` or `npm`:
+
+```bash
+npm install -S maplibre-gl maplibre-gl-js-amplify
+```
+
+Create a map onto which you can add the maplibre-gl-geocoder as a [control](https://maplibre.org/maplibre-gl-js-docs/api/markers/#icontrol). See the [documentation](~/lib/geo/maps.md) on creating and displaying maps. 
+
+maplibre-gl-geocoder requires a geocoding API for executing the search requests. You can easily define a geocoding API that wraps the [Amplify Geo APIs](#search-apis). Finally add the geocoder control to the map.
+
+```javascript
+import Amplify, { Geo } from "aws-amplify";
+import { AmplifyMapLibreRequest } from "maplibre-gl-js-amplify";
+import maplibregl, { Map } from "maplibre-gl";
+import MaplibreGeocoder from "@maplibre/maplibre-gl-geocoder";
+import "@maplibre/maplibre-gl-geocoder/dist/maplibre-gl-geocoder.css";
+import awsconfig from './aws-exports';
+
+Amplify.configure(awsconfig);
+
+async function initializeMap() {
+    const el = document.createElement("div");
+    el.setAttribute("id", "map");
+    document.body.appendChild(el);
+
+    const map = await AmplifyMapLibreRequest.createMapLibreMap({
+        container: "map",
+        center: [-123.1187, 49.2819],
+        zoom: 11,
+        region: "us-west-2"
+    })
+
+    // Define a geocoderApi to be used by `MaplibreGeocoder` that wraps the Amplify Geo APIs
+    const geocoderApi = {
+        forwardGeocode: async (config) => {
+            const data = await Geo.searchByText(config.query);
+
+            const features = data.map((result) => {
+                const { geometry, ...otherResults } = result;
+                return {
+                    type: "Feature",
+                    geometry: { type: "Point", coordinates: geometry.point },
+                    properties: { ...otherResults },
+                    place_name: otherResults.label,
+                    text: otherResults.label,
+                    center: geometry.point,
+                };
+            });
+            return { features }; // Must return an object with an array of features
+        }
+    };
+
+    const geocoder = new MaplibreGeocoder(geocoderApi, {
+        maplibregl: maplibregl,
+        showResultMarkers: true,
+    });
+    map.addControl(geocoder);
+}
+
+initializeMap();
+```
+
+![A search box on the top right corner of a map](~/images/geocoder-search-box-map.png)
+
+### Display the location search box outside the map
+
+You can also use [maplibre-gl-geocoder](https://github.com/maplibre/maplibre-gl-geocoder) to display the search box outside of the map and anywhere in your application. Instead of adding as a `control` on the maplibre map, extract the html element using function `onAdd()` and attach it anywhere in your DOM.
```

Comment:
```suggestion
You can also use [maplibre-gl-geocoder](https://github.com/maplibre/maplibre-gl-geocoder) to display the location search UI component anywhere in your application, even outside the map.

To do so, extract the html element using function `onAdd()` and attach it anywhere in your DOM instead of adding it via the map's `addControl` function.
```